DB/src/main/java/io/deephaven/db/tables/TableDefinition.java

```diff
@@ -577,20 +577,19 @@ private static void persistDefinitionImpl(@NotNull final TableDefinition definit
      * Persist the supplied TableDefinition to the default file for the supplied directory and table name.
      * @param definition The definition
      * @param destinationDirectory The destination directory
-     * @param tableName The table name
      */
-    public static void persistDefinition(@NotNull final TableDefinition definition, @NotNull final File destinationDirectory, @NotNull final String tableName) {
+    public static void persistDefinition(@NotNull final TableDefinition definition, @NotNull final File destinationDirectory) {
         persistDefinitionImpl(definition, new File(destinationDirectory, DEFAULT_FILE_NAME));
     }
 
     /**
      * Load a TableDefinition from the supplied file.
-     * @param source The file to find the definition in
+     * @param sourceDirectory The directory to find the definition in
      * @return The TableDefinition object
      * @throws UncheckedIOException or other RuntimeException if the file is not found or read fails
      */
-    public static TableDefinition loadDefinition(@NotNull final File source) {
-        return loadDefinitionImpl(source, false);
+    public static TableDefinition loadDefinition(@NotNull final File sourceDirectory) {
```

Comment:
It's not obvious to me that this is a good change. We should support persisting-to and loading-from a named file, rather than specifying a directory and inferring a filename. The latter is necessary in a DHE system, but we should only expose the former primitive here.

Reply:
Done, no more writing of table def.

Reply:
I will move everything (readTable and writeTable on TMT) to filenames; it is going to be a rather spread-out change tho, so will take me a bit.

Reply:
I implemented our last discussion, detecting via extension.